src/pineko/kfactor.py

```diff
@@ -0,0 +1,528 @@
+"""Module to include k_factors in grids."""
+import io
+import pathlib
+
+import numpy as np
+import pineappl
+import rich
+import yaml
+
+from . import scale_variations
+
+DEFAULT_PDF_SET = "NNPDF40_nnlo_as_01180"
+
+
+def read_kfactor(kfactor_path):
+    """Read the k_factor and returns the central values and the pdfset used to compute it."""
+    with open(kfactor_path, encoding="utf-8") as f:
+        stars = f.readline()
+        if not stars.startswith("*"):
+            raise TypeError("First line should start with '*'.")
+        descstring = io.StringIO()
+        for line in f:
+            if line.startswith("*"):
+                break
+            descstring.write(line)
+        description = descstring.getvalue()
+        try:
+            data = np.loadtxt(f)
+        except Exception as e:
+            raise TypeError(e) from e
+        data = data.reshape(-1, 2)
+        central_value = data[:, 0]
+        pdf_set = description.split(sep="PDFset:")[-1].split(sep="\n")[0].strip()
```

Comment:
Not all pdfset names are in lhapdf format, see e.g. seaquest where it's just "NNPDF3.1 NNLO". Also in many cases the pdfs will not be in lhpadf so they will first have to be obtained through vp.

Reply:
I understand what you tried, but I think in this case you'd end up doing `lhapdf.mkAlphaS("NNPDF3.1 NNLO")`, since `pdf_set` here would not be an empty string as you check below. 

My suggestion would be to just remove the if statement check below to really make this function called `read_kfactor` about only reading the cfactor file. Then below you can do a try except when doing `lhapdf.mkAlphaS`.

Reply:
I agree and I can even do it but, as I said, I believe that just editing the K-factor file is better and it is what I did. I think that in general it does not make much sense that the PDF written there is not an actual PDF so I don't think catching the error  is the best strategy in this case

Reply:
I see, so you want it to raise an error if there is a `PDFset` that is not in lhapdf, but use NNPDF4.0 if there is no `PDFset` at all? That's might be a reasonable enough thing to do, but it would be good if that were reflected more explicitly in the code as well.

Reply:
Yes you are right and yes, that is the idea